### PR TITLE
✨ Add certmanager flavor

### PR DESCRIPTION
**Pull request by charleszheng44**
**What this PR does / why we need it**:
Use cert-manager to manage the PKI for the NCP components.

### CODE REVIEW

File: templates/cluster-template-certmanager.yaml [hunk 1]
```diff
@@ -0,0 +1,293 @@
...
+---
+### apiserver-client ###
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: "${CLUSTER_NAME}-apiserver-client"
+spec:
+  secretName: "${CLUSTER_NAME}-apiserver-client"
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
```
Comment (christopherhein) [inline templates/cluster-template-certmanager.yaml:114]:
```suggestion
  duration: 8760h # 1y
  renewBefore: 4380h # 6m
```

To mirror CAPI and the default setup can we set this to 1 year and 6months?

File: templates/cluster-template-certmanager.yaml [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+---
+### etcd-client 
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: "${CLUSTER_NAME}-etcd-client"
+spec:
+  secretName: "${CLUSTER_NAME}-etcd-client"
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
```
Comment (christopherhein) [inline templates/cluster-template-certmanager.yaml:198]:
```suggestion
  duration: 8760h # 1y
  renewBefore: 4380h # 6m
```

File: templates/cluster-template-certmanager.yaml [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+---
+### proxy-client
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: "${CLUSTER_NAME}-proxy-client"
+spec:
+  secretName: "${CLUSTER_NAME}-proxy-client"
+  duration: 2160h # 90d
+  renewBefore: 360h # 15d
```
Comment (christopherhein) [inline templates/cluster-template-certmanager.yaml:281]:
```suggestion
  duration: 8760h # 1y
  renewBefore: 4380h # 6m
```

**Review by christopherhein — CHANGES_REQUESTED** [review]
Overall looks good but I think we need to set longer expiries on the client certs like the default config.

**Comment by charleszheng44** [comment]
> Overall looks good but I think we need to set longer expiries on the client certs like the default config.

Just prolong the expiries.

**Review by christopherhein — APPROVED** [review]
🎉 

Nice work!
/lgtm

**Pull request merged**